File: 卡码/代码随想录/C++/二叉树/02-二叉树属性/LeetCode_0404.cpp
//404. ×óÒ¶×ÓÖ®ºÍ
#include <iostream>
#include "../TreeNode.h"
using namespace std;

class Solution {
public:
  int sumOfLeftLeaves(TreeNode* root) {
		if(!root) return 0;
		int leftSum=0;
		if(root->left&&!root->left->left&&!root->left->right) leftSum=root->left->val;
		return leftSum+sumOfLeftLeaves(root->left)+sumOfLeftLeaves(root->right);
  }
};

int main(){
  system("pause");
  return 0;
}

File: 卡码/代码随想录/Java/codeThoughts/src/main/java/com/kurumi/leetCode/codeThoughts/二叉树/_02_二叉树属性/LeetCode_0404.java
package com.kurumi.leetCode.codeThoughts.二叉树._02_二叉树属性;

import com.kurumi.leetCode.codeThoughts.entity.TreeNode;

/**
 * @Author kurumi
 * @Date 2026-07-03 上午11:52:20
 * @Description 404. 左叶子之和
 */
public class LeetCode_0404 {
    static class Solution {
        public int sumOfLeftLeaves(TreeNode root) {
            if (root == null)
                return 0;
            int leftSum = 0;
            if (root.left != null && root.left.left == null && root.left.right == null)
                leftSum = root.left.val;
            return leftSum + sumOfLeftLeaves(root.left) + sumOfLeftLeaves(root.right);
        }
    }
}
